packages/liveblocks-core/src/client.ts

```diff
@@ -248,24 +275,100 @@ export function createClient(options: ClientOptions): Client {
       newRoom.connect();
     }
 
-    return newRoom;
+    roomIdsByTicket.set(ticket, roomId);
+    return ticket;
+  }
+
+  function enterRoom<
+    TPresence extends JsonObject,
+    TStorage extends LsonObject = LsonObject,
+    TUserMeta extends BaseUserMeta = BaseUserMeta,
+    TRoomEvent extends Json = never,
+  >(
+    roomId: string,
+    options: EnterOptions<TPresence, TStorage>
+  ): {
+    room: Room<TPresence, TStorage, TUserMeta, TRoomEvent>;
+    leave: () => void;
+  } {
+    const ticket = createTicketForRoom(roomId, options);
```

Comment:
Implementation details, so not blocking at all. Feel free to merge as is.

Is it possible to simplify the implementation by storing a map of roomId and array of leave callbacks instead of tickets `Map<string, Array<() => void>>`? What we're trying to do here is close the `EventSource` pattern so my intuition is telling me it could be modeled in the same way.

Reply:
Hmm. Maybe yeah, I'd like that simplification. I'll give it a try!

Reply:
Look at this [beautiful cleanup of the internals](https://github.com/liveblocks/liveblocks/commit/bf55fb4a73421a4c0f8ef7e999fd30ead425db74) — thanks for the suggestion! ❤️